lib/logstash/inputs/imap.rb

```diff
@@ -81,10 +83,23 @@ def check_mail(queue)
       items.each do |item|
         next unless item.attr.has_key?("RFC822")
         mail = Mail.read_from_string(item.attr["RFC822"])
-        queue << parse_mail(mail)
+        if @strip_attachments
+          queue << parse_mail(mail.without_attachments!)
+        else
+          queue << parse_mail(mail)
+        end
       end
 
       imap.store(id_set, '+FLAGS', @delete ? :Deleted : :Seen)
+      
+      if @delete_expunge_on_process_msg 
+        # Force messages to be marked as "Deleted", the above may or may not be working as expected. "Seen" means nothing if you are going to
+        # delete a message after processing.
+        
+        imap.store(id_set, '+FLAGS', [:Deleted])
```

Comment:
I didn't want to modify the existing Delete logic (that didn't seem to work)...for our setup at least, so I wanted to make my 'new method' independent of what was already going on.